### ISSUE TITLE
Type object 'datetime.datetime' has no attribute 'fromisoformat'

### ISSUE THREAD

**Issue by rjdmartins** [CS-SI/eodag-sentinelsat#21]
**Describe the bug**
When doing a eodag search by scihub provider, throws an AttributeError: type object 'datetime.datetime' has no attribute 'fromisoformat'.

**Code To Reproduce**
```py
from eodag.utils.logging import setup_logging
from eodag.api.core import EODataAccessGateway

import os
import traceback

def search(providerPriorityList):
    try:
        setup_logging(verbose=1)
        workspace = os.getcwd()
        print(workspace)
        dag = EODataAccessGateway(user_conf_file_path=os.path.join(workspace, 'user_conf_template.yml'))
        
        extent = { 'latmin': 37.249188, 'latmax': 37.299456, 'lonmin': -8.29536, 'lonmax': -8.404541 }
        start='2016-08-01'
        end='2018-12-01'

        defaultSatellite = 'Sentinel2'
        defaultLevel = 'Level2A'
        d ={
            'Sentinel1':{
                'GRD': {
                    'peps':'S1_SAR_GRD',
                    'scihub':'S1_SAR_GRD'
                },
                'OCN': {
                    'peps':'S1_SAR_OCN',
                    'scihub':'S1_SAR_OCN',
                },
                'SLC': {
                    'peps':'S1_SAR_SLC',
                    'scihub':'S1_SAR_SLC',
                },
            },
            'Sentinel2':{
                'Level1C':{
                    'peps':'S2_MSI_L1C',
                    'scihub':'S2_MSI_L1C',
                },
                'Level2A':{
                    'peps':'S2_MSI_L2A', 
                    'theia':'S2_MSI_L2A_MAJA',
                    'scihub':'S2_MSI_L2A',
                },
            }
        }

        providersAndProductTypes = d[defaultSatellite][defaultLevel]
        products = []
        for ppl in providerPriorityList:
            provider = ppl
            print("======== "+provider+" ============")
            if ppl in providersAndProductTypes:
                productType = providersAndProductTypes[ppl]
                print("========OK! Search "+provider+" in "+productType+"========")
                dag.set_preferred_provider(provider)
                products += dag.search_all(
                    productType=productType,
                    start=start, end=end,
                    geom=extent
                )                
            else:
                print("========ERROR! "+provider+" not found for "+defaultSatellite+" "+defaultLevel+"========")
                #exit(1)
        
        #generate json with the needed information
        print(len(products))
        for p in products:
            pjson = p.as_dict()
            print('------')
            #online, staging, offline
            customOutput = {
                'uid': pjson['properties']['uid'], 
                'filename': pjson['properties']['title'],
                'download': pjson['properties']['downloadLink'],
                'quicklook': pjson['properties']['quicklook'],
                'thumbnail': pjson['properties']['thumbnail'],
                'date': pjson['properties']['startTimeFromAscendingNode'],
                'storageStatus': pjson['properties']['storageStatus']
            }
            print(customOutput)
        
        print('Finished!')
        exit(0)
    except Exception as e:
        print(e)
        print(traceback.format_exc())
        exit(1)

try:
    providerPriorityList = ['scihub'] 
    search(providerPriorityList)
    exit(0)
except Exception as e:
    print(e)
    print(traceback.format_exc())
    exit(1)
```

**Output**
```
2021-05-05 17:01:28,674-15s eodag.config                     [INFO    ] Loading user configuration from: /home/ricardo/Worsica/user_conf_template.yml
2021-05-05 17:01:28,711-15s eodag.config                     [INFO    ] scihub: unknown provider found in user conf, trying to use provided configuration
2021-05-05 17:01:33,053-15s eodag.core                       [INFO    ] Locations configuration loaded from /home/ricardo/.config/eodag/locations.yml
======== scihub ============
========OK! Search scihub in S2_MSI_L2A========
2021-05-05 17:01:39,765-15s eodag.core                       [INFO    ] Searching product type 'S2_MSI_L2A' on provider: scihub
2021-05-05 17:01:39,766-15s eodag.core                       [INFO    ] Searching product type 'S2_MSI_L2A' on provider: scihub
2021-05-05 17:01:39,771-15s eodag.core                       [INFO    ] No result from provider 'scihub' due to an error during search. Raise verbosity of log messages for details
type object 'datetime.datetime' has no attribute 'fromisoformat'
Traceback (most recent call last):
  File "./eodag_search.py", line 60, in search
    geom=extent
  File "/home/foobar/py363_venv/lib/python3.6/site-packages/eodag/api/core.py", line 759, in search_all
    **kwargs,
  File "/home/foobar/py363_venv/lib/python3.6/site-packages/eodag/api/core.py", line 632, in search_iter_page
    search_plugin, count=False, raise_errors=True, **search_kwargs
  File "/home/foobar/py363_venv/lib/python3.6/site-packages/eodag/api/core.py", line 938, in _do_search
    res, nb_res = search_plugin.query(count=count, **kwargs)
  File "/home/foobar/py363_venv/lib/python3.6/site-packages/eodag_sentinelsat/eodag_sentinelsat.py", line 69, in query
    query_params, provider_product_type = self.update_keyword(**kwargs)
  File "/home/foobar/py363_venv/lib/python3.6/site-packages/eodag_sentinelsat/eodag_sentinelsat.py", line 256, in update_keyword
    datetime.fromisoformat(qp.pop("start")),
AttributeError: type object 'datetime.datetime' has no attribute 'fromisoformat'
```
**Environment:**
 - Python version: 3.6.3
 - EODAG version: 2.2.0
 - eodag-sentinelsat version: 0.3.0

**Additional context**
Doing a search for this error, this occurs because this function fromisoformat is only available from Python 3.7.
In Python 3.6.3, this function does not exist.
Although, there are some backports to fix this (https://github.com/movermeyer/backports.datetime_fromisoformat), but may not fix the problem, since this error occurs inside of the eodag_sentinelsat.py.
Thus, a workaround/fix for this is required for the people who run Python below 3.7. Maybe replace this function with strptime, like this fix? (https://github.com/precice/EventTimings/commit/7cf5b06693bef98486a5391e05358a543fab4e3e)

**Comment by maximlt** [comment]
Thanks for reporting this bug. `eodag` relies on `dateutils` instead of this "new" Python method to do the same thing (convert an ISO formatted datetime string to a datetime object). See this snippet that could be partially reused here:
https://github.com/CS-SI/eodag/blob/9096a12402068ac711b825956166a3de1051823f/eodag/api/product/metadata_mapping.py#L255-L267

PRs are welcome if you feel like so :)

**Comment by ahuarte47** [comment]
Hi, [backports-datetime-fromisoformat](https://pypi.org/project/backports-datetime-fromisoformat/) could be useful as workaround:

You can inject this method in your app using:

```
# >=Python 3.7 therefore 'datetime.date' defines 'fromisoformat'?
if not hasattr(datetime.date, 'fromisoformat'):
    # datetime.fromisoformat is only available in Python 3.7
    # Use the back port if this function is not detected.
    # https://pypi.org/project/backports-datetime-fromisoformat/
    #
    exec('from backports.datetime_fromisoformat import MonkeyPatch')
    exec('MonkeyPatch.patch_fromisoformat()')

```

**Comment by sbrunato** [comment]
> `eodag` relies on `dateutils` instead of this "new" Python method to do the same thing (convert an ISO formatted datetime string to a datetime object). 

Yes @maximlt , in `eodag-sentinelsat` we have to use `dateutil.parser.isoparse` instead of `datetime.fromisoformat` the same way we do in `eodag`.

Thanks @rjdmartins for reporting this issue
